numba/tests/test_dyn_array.py

```diff
@@ -600,6 +600,7 @@ def func(arr):
             return pyfunc(arr)
         self.check_like(func, np.float64)
 
+    @unittest.skipIf(numpy_version <= (1, 6), "test fails on Numpy 1.6")
```

Comment:
Actually, it's both :)
It turns out I can fix the test for empty_like(), but Numpy's zeros_like() doesn't work on 1.6 with a structured dtype...
Will commit a fix.